### PR TITLE
Add some status to CHI

### PR DESCRIPTION
**Pull request by su-houzhen**
### What type of PR is this?
/enhancement

### Which issue(s) this PR fixes?
N/A

### What this PR does?
Summary:
Add status `Creating` `Running` `CreateFailed` `UpdateFailed` to chi.

### Special notes for your reviewer?

### CODE REVIEW

File: pkg/apis/clickhouse.radondb.com/v1/type_status.go [hunk 1]
```diff
@@ -69,6 +73,20 @@ func (s *ChiStatus) ReconcileComplete() {
 	s.Action = ""
 }
 
+func (s *ChiStatus) Running() {
+	s.Status = StatusRunning
+	s.Action = ""
+}
+
+func (s *ChiStatus) ReconcileFailed(isUpdate bool) {
```
Comment (TCeason) [inline pkg/apis/clickhouse.radondb.com/v1/type_status.go:81]: Why this func needs a param?

File: pkg/controller/chi/worker.go [hunk 2]
```diff
@@ -384,6 +387,27 @@ func (w *worker) updateCHI(old, new *chop.ClickHouseInstallation) error {
...
+		if err != nil {
+			w.a.Error("ERROR ping on host %s. err: %v", host.Name, err)
+			return err
+		}
+		return nil
+	})
+
+	for _, value := range res {
+		if value != nil {
+			(&new.Status).ReconcileFailed(update)
```
Comment (TCeason) [inline pkg/controller/chi/worker.go:402]: The value of `update` will be modified in two (&new.Status).ReconcileFailed(update) ref?

https://github.com/radondb/clickhouse-operator/pull/1/files#diff-4bd0c369f4915649d774821b536e0966416409ee829fa19be9bfd48a092b9281R326

https://github.com/radondb/clickhouse-operator/pull/1/files#diff-4bd0c369f4915649d774821b536e0966416409ee829fa19be9bfd48a092b9281R402

File: pkg/apis/clickhouse.radondb.com/v1/type_status.go (hunk 1 above)
Comment (su-houzhen) [reply pkg/apis/clickhouse.radondb.com/v1/type_status.go:81]: judge `createfailed` or `updatefailed`

Comment (TCeason) [reply pkg/apis/clickhouse.radondb.com/v1/type_status.go:81]: I think it's better to add new reconciles.

Comment (TCeason) [reply pkg/apis/clickhouse.radondb.com/v1/type_status.go:81]:
```
StatusCreating     = "Creating"
StatusCreateFailed = "CreateFailed"
StatusUpdateFailed = "UpdateFailed"
StatusTerminating  = "Terminating"
```

In my mind if you add four states, maybe should add four reconciles.

**Pull request closed without merging**